[receiver/sshcheck] Flaky test TestScraper/metrics_golden_sftp

### Component(s)

receiver/sshcheck

### What happened?

```
--- FAIL: TestScraper (4.80s)
    --- FAIL: TestScraper/metrics_golden (0.02s)
    --- FAIL: TestScraper/metrics_golden_sftp (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf58549]

goroutine 65 [running]:
testing.tRunner.func1.2({0x118dda0, 0x1a3cc70})
	/opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1389 +0x366
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1392 +0x5d2
panic({0x118dda0, 0x1a3cc70})
	/opt/hostedtoolcache/go/1.18.10/x64/src/runtime/panic.go:844 +0x258
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver/internal/configssh.(*Client).SFTPClient(0xc00010eb40)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sshcheckreceiver/internal/configssh/configssh.go:68 +0x49
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver.(*sshcheckScraper).scrapeSFTP(0xc00043a0a0, 0x173f3ebb84946e6f?)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sshcheckreceiver/scraper.go:69 +0x7b
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver.(*sshcheckScraper).scrape(0xc00043a0a0, {0x1441d60, 0xc00003c128})
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sshcheckreceiver/scraper.go:133 +0x5d8
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver.TestScraper.func2(0x0?)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sshcheckreceiver/scraper_test.go:132 +0x925
testing.tRunner(0xc0000b0ea0, 0xc000339110)
	/opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1439 +0x214
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1486 +0x725
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver	4.860s
```

Seen here: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4049569807/jobs/6966104240

### Collector version

current main

### Environment information

_No response_

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

--- a/receiver/sshcheckreceiver/internal/configssh/configssh.go
+++ b/receiver/sshcheckreceiver/internal/configssh/configssh.go
@@ -65,7 +65,7 @@ func (c *Client) Dial(endpoint string) (err error) {
 }
 
 func (c *Client) SFTPClient() (*SFTPClient, error) {
-	if c.Client.Conn == nil {
+	if c.Client == nil || c.Client.Conn == nil {
 		return nil, fmt.Errorf("SSH client not initialized")
 	}
 	client, err := sftp.NewClient(c.Client)

--- a/receiver/sshcheckreceiver/scraper_test.go
+++ b/receiver/sshcheckreceiver/scraper_test.go
@@ -37,7 +37,6 @@ type opensshContainer struct {
 }
 
 func setupSSHServer(t *testing.T) *opensshContainer {
-	ctx := context.Background()
 	wd, err := os.Getwd()
 	require.NoError(t, err)
 	req := testcontainers.ContainerRequest{
@@ -56,18 +55,19 @@ func setupSSHServer(t *testing.T) *opensshContainer {
 		ExposedPorts: []string{"2222/tcp"},
 		WaitingFor:   wait.ForExposedPort(),
 	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*60)
+	defer cancel()
+
 	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
 		ContainerRequest: req,
 		Started:          true,
 	})
 	require.NoError(t, err)
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
-	defer cancel()
-
 	endpoint, err := container.PortEndpoint(ctx, "2222", "")
 	require.NoError(t, err)
-
+	t.Log("endpoint: ", endpoint)
 	return &opensshContainer{Container: container, Endpoint: endpoint}
 }
 
@@ -75,7 +75,6 @@ func TestScraper(t *testing.T) {
 	if !supportedOS() {
 		t.Skip("Skip tests if not running on one of: [linux, darwin, freebsd, openbsd]")
 	}
-	t.Parallel()
 	c := setupSSHServer(t)
 	defer func() {
 		require.NoError(t, c.Terminate(context.Background()), "terminating container")
@@ -151,7 +150,6 @@ func TestScraperDoesNotErrForSSHErr(t *testing.T) {
 	if !supportedOS() {
 		t.Skip("Skip tests if not running on one of: [linux, darwin, freebsd, openbsd]")
 	}
-	t.Parallel()
 	c := setupSSHServer(t)
 	defer func() {
 		require.NoError(t, c.Terminate(context.Background()), "terminating container")
@@ -179,7 +177,6 @@ func TestTimeout(t *testing.T) {
 	if !supportedOS() {
 		t.Skip("Skip tests if not running on one of: [linux, darwin, freebsd, openbsd]")
 	}
-	t.Parallel()
 	testCases := []struct {
 		name     string
 		deadline time.Time
@@ -230,4 +227,26 @@ func TestCancellation(t *testing.T) {
 	require.Error(t, err, "should have returned error on canceled context")
 	require.EqualValues(t, err.Error(), ctx.Err().Error(), "scrape should return context's error")
 
+}
+
+// issue # 18193
+// init failures resulted in scrape panic for SFTP client
+func TestWithoutStartErrsNotPanics(t *testing.T) {
+	f := NewFactory()
+	cfg := f.CreateDefaultConfig().(*Config)
+	cfg.ScraperControllerSettings.CollectionInterval = 100 * time.Millisecond
+	cfg.Username = "otelu"
+	cfg.Password = "otelp"
+	cfg.Endpoint = "localhost:22"
+	cfg.IgnoreHostKey = true
+	cfg.Metrics.SshcheckSftpStatus.Enabled = true
+	cfg.Metrics.SshcheckSftpDuration.Enabled = true
+
+	// create the scraper without starting it, so Client is nil
+	scrpr := newScraper(cfg, receivertest.NewNopCreateSettings())
+
+	// scrape should error not panic
+	var err error
+	require.NotPanics(t, func() { _, err = scrpr.scrape(context.Background()) }, "scrape should not panic")
+	require.Error(t, err, "expected scrape to err when without start")
 }

